src/haproxy/conn.go

```diff
@@ -0,0 +1,126 @@
+package haproxy
+
+import (
+	"bufio"
+	"net"
+	"time"
+
+	proxyproto "github.com/pires/go-proxyproto"
+)
+
+// Conn is a `net.Conn` compatible struct that handles PROXY header checking.
+type Conn struct {
+	rd  *bufio.Reader
+	c   net.Conn
+	r   net.Addr
+	l   net.Addr
+	hdr *proxyproto.Header
+}
+
+// NewConn wraps a connection with a `bufio.Reader` for checking if the
+// PROXY protocol headers are present
```

Comment:
Docs are a bit misleading here, the user never sees a `bufio.Reader`.  Perhaps something like

> NewConn returns a connection that parses PROXY protocol headers from the start of the stream.

Or something like that.

Reply:
Also fixed in the next commit 👍